pkg/apis/overcommit/v1alpha1/types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeOvercommitConfigSpec is a description of a NodeOvercommitConfig
+type NodeOvercommitConfigSpec struct {
+	// NodeOvercommitSelectorVal is the value of node label selector with key consts.NodeOvercommitSelectorKey,
+	// it decides whether to update Nodes if the Node matches the selector 'consts.NodeOvercommitSelectorKey=NodeOvercommitSelectorVal'
+	// +optional
+	NodeOvercommitSelectorVal string `json:"nodeOvercommitSelectorVal,omitempty"`
+
+	// ResourceOvercommitRatioConfig describes the resource overcommit ratio that needs to inject into Node.Annotations
+	// cpu,memory are supported.
+	// +optional
+	ResourceOvercommitRatioConfig map[v1.ResourceName]string `json:"resourceOvercommitRatioConfig,omitempty"`
```

Comment:
why not use a slice of custom defined structure with resource name as a field instead of map, because the scalebility of map is poor, and k8s also suggest using list instead of map [https://github.com/kubernetes/kubernetes/issues/2004#issuecomment-60641437](url)

Reply:
> why not use a slice of custom defined structure with resource name as a field instead of map, because the scalebility of map is poor, and k8s also suggest using list instead of map [https://github.com/kubernetes/kubernetes/issues/2004#issuecomment-60641437](url)

thanks for the suggestion. The scalebility is pool that each resource only has a string value. I'll update it accroding to the suggestion.

Reply:
@luomingmeng  I thought about it, maybe "ResourceOvercommitRatio" is more like "ResourceList" in container Resources, also we need a map to ensure the uniqueness of the resources. What do you think?
Do you have any suggestions? @caohe

Reply:
If the field is similar to `ResourceList`, it is better to use the name `ResourceOvercommitRatio` directly, rather than using `ResourceOvercommitRatioConfig`.

Reply:
done, ResourceOvercommitRatioConfig is renamed to ResourceOvercommitRatio.